### ISSUE TITLE
Weblate: string updated in the English source files should be reset in all other languages

### ISSUE THREAD

**Issue by paglias** [HabitRPG/habitica#12126]
Similar to #12119

When a string is updated in the English source files they should be reset to the english version in all other languages since the meaning might have changed and the translation be outdated.

We might have to change a setting in Weblate or update the script that updates the translation files here.

----------

The only solution might be to use Weblate's API https://docs.weblate.org/en/latest/api.html or

> Sounds like the best approach for our current set of tools would be to create a script that:
> 
> * Either takes a user-specified list of string tokens...
> * ...or finds them by looking at the differences in `en` files over a commit
> * Then rewrites all the corresponding strings in non-`en` locale files to the new English text (removing those removed from the english text)
> 
> We'll then run that script, commit, and merge to Weblate as part of our regular Weblate repo maintenance (a couple times a week, with deploys). Weblate users will need to look back at the history to see previous translations to help them rewrite, but it's probably the only way with Weblate's feature set for us to be sure translated text isn't showing outdated information.
>

**Comment by phillipthelen** [comment]
I could not find a way to configure this in weblate unfortunately. So we'd have to change the source/translation files.

**Comment by paglias** [comment]
We should probably write a script or change the existing one that uploads new translations to weblast. @phillipthelen @SabreCat do you know where the code for that is?

**Comment by SabreCat** [comment]
Hmm, @gryphonbot does that, but I don't know how it's configured...

**Comment by SabreCat** [comment]
Oh, now that I think about it, it's all Git-based. Weblate pulls `develop`, and builds out locale strings based on our `en` tokens plus the list of supported languages. Then commits the translated strings and pushes to develop via @gryphonbot.

**Comment by paglias** [comment]
thanks! i thought it would be in this repo https://github.com/HabitRPG/Habitica-Webhooks but it's not so it might be on the VM with the remaining scripts that also post to slack?

**Comment by SabreCat** [comment]
No, I don't think we have our own script. It's just Weblate, given creds for a GitHub account. (or API key, more likely, ykwim)

**Comment by paglias** [comment]
Ahh ok, that makes sense! Then I guess we should be able to setup something with a github action that only runs on those commits

**Comment by fmaximus** [comment]
Weblate has quite good support for updates in source strings.
Those are automatically counted under "Things to check",
with a diff style source change.
If only one word is changed I would rather keep the translation,
than to have to re-translate everything.
Especially if it's a only whitespace change.

**Comment by Alys** [comment]
@fmaximus Unfortunately for a lot of languages we've been seeing that strings are not being adjusted when the English version changes, and sometimes those adjustments are quite important, even those that are single-word changes sometimes. For example, we've seen cases where the end date of a limited availability item/event wasn't re-translated, meaning that people using non-English languages are shown misleading information.

However if you have any ideas for making the re-translation process more reliable without the need to delete previously translated versions of the strings, please tell us!

**Comment by fmaximus** [comment]
If the translation history in Weblate stays intact, I'm good.
In that way you could always copy the previous translation back.

Whitespace or punctuation only changes should not revert.

IMHO, dates should be handled by variables, and filled up with other translations, like .dateEndApril and .augustYYYY, or generated by moment. In this way the update is done programmatically, 
and all languages are immediately correct.

**Comment by SabreCat** [comment]
Sounds like the best approach for our current set of tools would be to create a script that:

* Either takes a user-specified list of string tokens...
* ...or finds them by looking at the differences in `en` files over a commit
* Then rewrites all the corresponding strings in non-`en` locale files to the new English text

We'll then run that script, commit, and merge to Weblate as part of our regular Weblate repo maintenance (a couple times a week, with deploys). Weblate users will need to look back at the history to see previous translations to help them rewrite, but it's probably the only way with Weblate's feature set for us to be sure translated text isn't showing outdated information.

**Comment by paglias** [comment]
@SabreCat I'm going to add your comment to the issue description since it's far more clear that what I wrote originally!

**Comment by LioZam** [comment]
I would like to check whether I can help with this issue. 

What can I do to investigate/fix this? Oh.. right... reading the guide lines first this time... but I wanted to at least tell you that I would be interested in looking in the git vs weblate issue(s) (this, as well as #12119 )

**Comment by paglias** [comment]
@LioZam I think the best approach is to follow the description in the first comment where it's suggested to create a script to solve this issue. To be honest I don't think this is a super easy issue to work on though

**Comment by LioZam** [comment]
yeah.. since I'll start work in a few days, I guess I shouldn't take on a big project. if it was just altering some scripts it was no problem.. but since it isn't even clear (for me) where what happens (e.g. if something in weblate could be changed, or if there are skripts, and if the latter how to access them/where to find them, and if the first, how to access that... oh well.. too many unknowns for a n00b here.) Hope y'all get this sorted out. _success!_ (a saying in Dutch equal to wishing someone "Good Luck")

**Comment by thewilloftheshadow** [comment]
I think I could take a shot at this. It looks pretty similar to things I’ve done before for my own repos, though I don’t use weblate myself

**Comment by paglias** [comment]
@thewilloftheshadow I'll mark this as in progress for you, thanks!

**Comment by thewilloftheshadow** [comment]
Is there some API in weblate that will show the recently updated strings, or do I need to go the route of comparing diffs to find changed strings

**Comment by paglias** [comment]
That would be a question for @SabreCat / @phillipthelen

**Comment by beffymaroo** [comment]
@SabreCat @phillipthelen  Per the above? @thewilloftheshadow Let me know if you'd like to work on this pending the answer from Sabe or Phillip, and if not I can adjust the issue labels accordingly. Thank you!

**Comment by thewilloftheshadow** [comment]
You can release this one, I don’t have time anymore sadly

**Comment by beffymaroo** [comment]
@thewilloftheshadow Thanks for replying! I'll put this issue back in Help Wanted

**Comment by SabreCat** [comment]
Weblate's documentation is here, for those interested in digging in: https://docs.weblate.org/en/latest/index.html I'll give @phillipthelen another ping to see if the answer is close at hand, though!

**Comment by phillipthelen** [comment]
I updated Weblate to the most recent version and also installed a new add-on that cleans up translation files. I'm not entirely sure if that handles the exact use case here, so we'll have to keep an eye on the changes and see

**Comment by SabreCat** [comment]
OK! I'll flip this to "investigation needed" and depending on how things shake out, we can open it back up for contributions or close it.
